cmdstanpy/utils/stancsv.py

```diff
@@ -129,245 +134,384 @@ def parse_hmc_adaptation_lines(
+    out: Dict[str, Union[str, int, float]] = {}
+    for key, val in extract_key_val_pairs(comment_lines):
+        if key == 'file':
+            if not val.endswith('csv'):
+                out['data_file'] = val
+        else:
+            if val == 'true':
+                out[key] = 1
+            elif val == 'false':
+                out[key] = 0
+            else:
+                for cast in (int, float):
+                    try:
+                        out[key] = cast(val)
+                        break
+                    except ValueError:
+                        pass
+                else:
+                    out[key] = val
+    return out
+
+
+def parse_header(header: str) -> Tuple[str, ...]:
+    """Returns munged variable names from a Stan csv header line"""
+    return tuple(munge_varname(name) for name in header.split(","))
+
+
+def construct_config_header_dict(
+    comment_lines: List[bytes], header: Optional[str]
+) -> Dict[str, Union[str, int, float, Tuple[str, ...]]]:
+    """Extracts config and header info from comment/draws lines parsed
+    from a Stan CSV file."""
+    config = parse_config(comment_lines)
+    out: Dict[str, Union[str, int, float, Tuple[str, ...]]] = {**config}
+    if header:
+        out["raw_header"] = header
+        out["column_names"] = parse_header(header)
+    return out
+
+
+def parse_variational_eta(comment_lines: List[bytes]) -> float:
+    """Extracts the variational eta parameter from stancsv comment lines"""
+    for i, line in enumerate(comment_lines):
+        if line.startswith(b"# Stepsize adaptation") and (
+            i + 1 < len(comment_lines)  # Ensure i + 1 is in bounds
+        ):
+            eta_line = comment_lines[i + 1]
+            break
+    else:
+        raise ValueError(
+            "Unable to parse eta from Stan CSV, adaptation block not found"
+        )
+
+    _, val = eta_line.split(b" = ")
+    return float(val)
+
+
+def extract_max_treedepth_and_divergence_counts(
+    header: str, draws_lines: List[bytes], max_treedepth: int, warmup_draws: int
+) -> Tuple[int, int]:
+    """Extracts the max treedepth and divergence counts from the header
+    and draw lines of the MCMC stan csv output."""
+    if len(draws_lines) <= 1:  # Empty draws
+        return 0, 0
+    column_names = header.split(",")
+
+    try:
+        indexes_to_keep = [
+            column_names.index("treedepth__"),
+            column_names.index("divergent__"),
+        ]
+    except ValueError:
+        # Throws if treedepth/divergent columns not recorded
+        return 0, 0
+
+    sampling_draws = draws_lines[1 + warmup_draws :]
+
+    filtered = filter_csv_bytes_by_columns(sampling_draws, indexes_to_keep)
+    arr = csv_bytes_list_to_numpy(filtered).astype(int)
+
+    num_max_treedepth = np.sum(arr[:, 0] == max_treedepth)
+    num_divergences = np.sum(arr[:, 1])
+    return num_max_treedepth, num_divergences
+
+
+def is_sneaky_fixed_param(header: str) -> bool:
```

Comment:
Fun name. Can you add a `# TODO: Remove after CmdStan 2.37 is the minimum version`?